component_contribution/training_data.py

```diff
@@ -1,297 +1,437 @@
+    that are used as training data.
+
+    Returns
+    -------
+    pandas.DataFrame
+
+    References
+    ----------
+    .. [1] Goldberg, Robert N., Yadu B. Tewari, and Talapady N. Bhat.
+           “Thermodynamics of Enzyme-Catalyzed Reactions—a Database for
+           Quantitative Biochemistry.” Bioinformatics 20, no. 16 (November 1,
+           2004): 2874–77. https://doi.org/10.1093/bioinformatics/bth314.
+    .. [2] http://xpdb.nist.gov/enzyme_thermodynamics/
+
+    """
+    columns = [
+        "url",
+        "reference",
+        "method",
+        "eval",
+        "ec_number",
+        "name",
+        "definition_kegg",
+        "description",
+        "standard_equilibrium_constant",
+        "equilibrium_constant",
+        "temperature",
+        "isoelectric_point",
+        "ph_value",
+        "log_magnesium_concentration"
+    ]
+    with open_binary(data, "TECRDB.tsv.gz") as archive:
+        with GzipFile(fileobj=archive) as file_handle:
+            table = pd.read_table(file_handle, sep="\t", header=None,
+                                  index_col=False, names=columns)
+    LOGGER.debug("Read %d rows.", len(table))
+    table["gibbs_energy_prime"] = \
+        -R * table["temperature"] * np.log(table["equilibrium_constant"])
+    return table
+    # [[
+    #     "reaction",
+    #     "delta_gibbs_zero",
+    #     "temperature",
+    #     "isoelectric_point",
+    #     "ph_value",
+    #     "log_magnesium_concentration"
+    #     "weight",
+    #     "is_balanced",
+    #     "reference",
+    #     "description"
+    # ]]
+
+
+def read_formations():
+    columns = [
+        "kegg_id",
+        "name",
+        "gibbs_energy_prime",
+        "ph_value",
+        "isoelectric_point",
+        "log_magnesium_concentration",
+        "temperature",
+        "decompose",
+        "reference",
+        "remark"
+    ]
+    # Replace the header by skipping the first row.
+    with open_text(data, "formation_energies_transformed.tsv") as file_handle:
+        table = pd.read_table(file_handle, sep="\t", header=None,
+                              skiprows=1, index_col=False, names=columns)
+    LOGGER.debug("Read %d rows.", len(table))
+    table["decompose"] = (table["decompose"] == 1)
+    return table
+
+
+def read_redox():
+    columns = [
+        "name",
+        "kegg_id_oxidized",
+        "number_protons_oxidized",
+        "charge_oxidized",
+        "kegg_id_reduced",
+        "number_protons_reduced",
+        "charge_reduced",
+        "nernst_potential_prime",
+        "ph_value",
+        "isoelectric_point",
+        "log_magnesium_concentration",
+        "temperature",
+        "reference"
+    ]
+    with open_text(data, "redox.tsv") as file_handle:
+        table = pd.read_table(file_handle, sep="\t", header=None,
+                              skiprows=1, index_col=False, names=columns)
+    LOGGER.debug("Read %d rows.", len(table))
+    return table
+
+
```

Comment:
Can you add a `TODO: refactor training data (do not remove)` so we don't forget why all of this is commented?

Reply:
Right, I put a comment on line 42 :stuck_out_tongue: makes more sense down there, though.